### PR TITLE
Reworking `nufmt`

### PR DESCRIPTION
**Pull request by AucaCoyan**
Hi! 

I still have to resolve some issues, and tests and other things, that's why this is still a draft

### CODE REVIEW

**Comment by fdncred** [comment]
Thanks. Looks good so far. Just ping me when you're ready for a review.

**Review by amtoine — COMMENTED** [review]
i've started having a little look :relieved: 

> **Note**
> files not reviewed at all
> - `formatting.rs`
> - `lib.rs`
> - `main.rs`

File: src/config.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    /// Default: 100
+    pub max_width: usize,
+    /// number of lines bafore and after a custom command
+    pub margin: usize,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Config {
+            tab_spaces: 2,
```
[the remark this answers is not in the document]
Comment (AucaCoyan) [reply src/config.rs:16]: Yes! I would too! The original project had 2 spaces, so I decided to stay with it. I use 4 spaces in everything.

File: examples/example.nu
[the remark this answers is not in the document]
Comment (AucaCoyan) [reply examples/example.nu:1]: I commited as is. The reality is that if you open with vs code the extension of nushell complains in a couple lines. I'm not sure what to do with it yet

File: src/config.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (amtoine) [reply src/config.rs:16]: that's ok to go away from the original project :relieved: 

you might want to look at the [specifications](https://github.com/fdncred/nufmt/blob/main/docs/specification.md) which are much more recent and should reflect more how Nushell *should be written* :wink:

File: examples/example.nu
[the remark this answers is not in the document]
Comment (amtoine) [reply examples/example.nu:1]: mm okok

we might want to have a closer look at it when we decide to use it :+1:

**Comment by fdncred** [comment]
There's also a work-in-progress style guide here https://github.com/nushell/nushell.github.io/pull/904

File: src/config.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/// options available to the formatter
+pub struct Config {
+    /// number of spaces of indent
+    pub tab_spaces: usize,
+    /// max amount of characters per line
+    /// # Maximum width of each line.
+    /// Default: 100
```
[the remark this answers is not in the document]
Comment (AucaCoyan) [reply src/config.rs]: With the new commit,  I decided to skip documenting the defaults. Resolved! ✅

File: src/config.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (amtoine) [reply src/config.rs]: noishe :yum:

**Comment by amtoine** [comment]
@AucaCoyan
is there at least an `nufmt` we can run?

if it's the case, i'd vote to land this and address the missing points in follow up PRs :relieved:

**Comment by fdncred** [comment]
I'm fine with landing this as a step in the right direction, with a mind to keep iterating on it until we get it to where we want it to be.

**Comment by AucaCoyan** [comment]
Ok! I will keep working, for sure!
I will mark this PR to review and submit issues to point the places to improve / fix

**Comment by fdncred** [comment]
nice initial work!

**Pull request merged**